esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -193,28 +195,21 @@ pub trait TransferTokensModule:
         args.push_arg(to);
         args.push_arg(mapped_tokens.len());
 
-        for token in &mapped_tokens {
-            args.push_arg(token.token_identifier);
-            args.push_arg(token.token_nonce);
-            args.push_arg(token.amount);
-        }
+        args.push_multi_arg(&mapped_tokens);
 
         args
     }
 
     #[promises_callback]
-    fn execute(
+    fn distribute_payments_callback(
```

Comment:
From what I remember this function name is actually the Identifier for the events in the function and we explicitly want `execute` the identifier. Please revert this change.
```suggestion
    fn execute(
```